app/controllers/api/v2/allocations_controller.rb

```diff
@@ -0,0 +1,33 @@
+module API
+  module V2
+    class AllocationsController < API::V2::ApplicationController
+      def create
+        authorize @allocation = Allocation.new(allocation_params.merge(accredited_body_id: accredited_body.id))
+
+        if @allocation.save
+          render jsonapi: @allocation, status: :created
+        else
+          render jsonapi_errors: @allocation.errors, status: :unprocessable_entity
+        end
+      end
+
+    private
+
+      def accredited_body
+        @accredited_body ||= begin
+                               accredited_body_code = params[:provider_code]
```

Comment:
If possible can we start using the term `training_provider(s)` as opposed to `provider` where appropriate 
See https://ukgovernmentdfe.slack.com/archives/CAHAJFR0X/p1586179199434100?thread_ts=1586179143.434000&cid=CAHAJFR0X

Reply:
I know it's possible as we are stuck with the routes and this is a more general point for wider discussion, but as an example this route that was added made my head do a couple of somersaults before I realised what was going on 

```
/api/v2/providers/:provider_code/allocations(.:format)      
```

Looking at the above it's not clear whether this is referring to the the Accrediting Body or Training Provider (yet appreciate these are both types of provider) - this knowledge is implicit, would be good if it were more explicit ?

Reply:
Yeah, I agree. I think we should consider using `accredited_body`, `training_provider` in the URLs too. I did discuss this a bit with @misaka when we were coming up with the URLs. Let's revisit as I think it's a good idea.

Reply:
We've discussed this further offline. If we change these URLs then we're really need to start considering `AccreditedBody` and `TraningProvider` as separate resources with the client classes etc in the frontend apps or we'd have to start making the existing resources quite messy to generate these URLs. So... I think we should move forward with this for now pending a bit of further discussion (cc/ @misaka)